packages/core/src/knapsack-pro-api.ts

```diff
@@ -41,7 +40,11 @@ export class KnapsackProAPI {
       node_index: KnapsackProEnvConfig.ciNodeIndex,
       node_build_id: KnapsackProEnvConfig.ciNodeBuildId,
       user_seat: KnapsackProEnvConfig.maskedUserSeat,
-      ...(shouldSendTestFilesInPayload && { test_files: allTestFiles }),
+      ...(isFirstRequest && {
```

Comment:
The `isFirstRequest` name could be misleading. We don't send test_files in the payload in the very first request.

First we attempt to fetch tests from the Queue. If the Queue is not initialised only then we attempt to initalize a new queue so in reality we send the test files in the payload on the 2nd requests or not at all (in case when we fetched tests on the 1st request).
See body examples in Queue API docs: https://docs.knapsackpro.com/api/v1/#queues_queue_post

I would probably stick to the old name `shouldSendTestFilesInPayload`.